File: assets/snippets/go/struct.go
package main

import "fmt"

type User struct {
    ID        int
    Name      string
    Email     string
    Age       int
}

func (u *User) Greet() string {
    return fmt.Sprintf("Hello, I'm %s", u.Name)
}

func main() {
    user := &User{
        ID:    1,
        Name:  "Alice",
        Email: "alice@example.com",
        Age:   30,
    }
    fmt.Println(user.Greet())
}

File: assets/snippets/rs/struct.rs
struct User {
    username: String,
    email: String,
    active: bool,
}

impl User {
    fn new(username: String, email: String) -> Self {
        Self {
            username,
            email,
            active: true,
        }
    }

    fn deactivate(&mut self) {
        self.active = false;
    }
}
